app/main.py

```diff
@@ -1,2 +1,63 @@
 class Dictionary:
-    pass
+    INITIAL_CAPACITY = 8
+    LOAD_FACTOR = 2 / 3
+    RESIZE = 2
+
+    class Node:
+        def __init__(self, key, value, key_hash):
+            self.key = key
+            self.value = value
+            self.key_hash = key_hash
+
+    def __init__(self):
+        self.container = [None] * self.INITIAL_CAPACITY
+        self.cells = 0
+        self.length = len(self.container)
+
+    def resize(self):
+        old_container = self.container
+        self.INITIAL_CAPACITY *= self.RESIZE
+        self.container = [None] * self.INITIAL_CAPACITY
+        self.length *= 2
+        self.cells = 0
+        for cell in old_container:
+            if cell:
+                self.__setitem__(cell.key, cell.value)
+
+    def __setitem__(self, key, value):
+        key_hash = hash(key)
+        index = key_hash % self.length
+
+        if self.cells == round(self.length * self.LOAD_FACTOR):
+            self.resize()
+
+        if not self.container[index]:
+            self.container[index] = self.Node(key, value, key_hash)
+            self.cells += 1
+
+        elif self.container[index].key == key:
```

Comment:
Maybe compare hashes first? It will be faster

Reply:
but I can have two equal hashes, with different keys

Reply:
Yes, but the comparison of the hashes works much faster (imagine the comparing time for two tuples of lengths 10^6). So you can compare hashes first which are two numbers and only when they are equal compare actual keys.
If the hashes are not equal you will save a lot of time :)